vignettes/date_time_datetime.Rmd

```diff
@@ -0,0 +1,51 @@
+
+Starting in **{datasetjson}** v0.3.0 we've introduce support for Dataset JSON v1.1.0. As such, we automatically handle date, time, and date time conversions. There are a few considerations you need to make when dealing with these types to make things work properly.
+
+# Metadata Settings
+
+Version 5 SAS Transport Files didn't have a notion of a "date", "time" or "datetime" type. Instead, using the SAS convention these were just Integer values with a display format attached. Dataset JSON Version 1.1 explicitly clarifies numeric date types using the `dataType` and `targetDataType` fields in the columns metadata. Consider these variables.
+
+```{r, echo=FALSE}
+tibble::tribble(
+  ~itemOID,             ~name,          ~label,             ~dataType, ~length,      ~targetDataType, ~displayFormat, ~keySequence,
+  'IT.DF.CHARDT',   'CHARDT',     'Character Date',     'date',   8L,          NA_character_,   NA_character_,  NA_integer_,
+  'IT.DF.CHARTM',   'CHARTM',     'Character Time',     'time',   10L,          NA_character_,   NA_character_,  NA_integer_,
+  'IT.DF.CHARDTM',   'CHARDTM',     'Character Datetime', 'datetime',   19L,         NA_character_,   NA_character_,  NA_integer_,
+  'IT.DF.NUMDT',   'NUMDT',     'Numeric date',     'time',     NA_integer_, "integer",       "TIME8",        NA_integer_,
+  'IT.DF.NUMTM',   'NUMTM',     'Numeric time',     'time',     NA_integer_, "integer",       "TIME8",        NA_integer_,
+  'IT.DF.NUMDTM',   'NUMDTM',     'Numeric datetime', 'datetime', NA_integer_, "integer",      "E8601DT",       NA_integer_
+) |>
+  kable()
+```
+
+In the table above, we have the metadata for both character and numeric dates, times, and date times. Both sets of variables have the same values within `dataType`. The difference is the optional field of `targetDataType`, where the value for the numeric variables is set to `integer`. Both `read_dataset_json()` and `write_dataset_json()` rely on these fields and as such they must be set properly. This comes with a few assumption and requirements.
+
+-   Numeric dates will be converted into the type of `Date`
+-   Numeric times will be converted to the **{lubridate}** type of `Period`
+  - R doesn't have a specific built in type of time. We decided to take on **{lubridate}** as a dependency given package stability and **{lubridate}**'s inclusion in the tidyverse. The `Period` objects on read are produced using the `lubridate::hms()`. 
+-   Numeric date times will be converted to the base R type of `POSIXlt`
+  - `POSIXlt` is used as opposed to `POSIXct` because we require that datetimes are set to the UTC timezone. CDISC dates are generally not timezone qualified, and unless a timezone is explicitly specified systems may default to the user's current timezone. As such, we've introduced a hard requirement that datetimes are anchored to UTC. If the datetime variable is found to be using a different timezone, an error will be thrown. 
```

Comment:
I'm not convinced yet why POSIXlt only is supported.  POSIXct addresses timezones as well.  One is stored in seconds, the other in S,M,H,D,Mon,Y.  If we have a POSIXct date/time, that is UTC, shouldn't we still be able to convert this to a character date?  Are we avoiding this because of the different reference dates (Jan 1, 1970 vs Jan 1, 1960)?

Reply:
I see the next sentence in this vignette is a disclaimer about these assumptions.  Maybe we just get it out like this so we can discuss further with the community and update.

Reply:
Well.  When I was looking yesterday I couldn't see the timezone attribute. I double checked and now I can. Let me revert to POSIXct because it's a less complex object. 

Sorry - let me revert it. This has a bit of a trickle effect.

Reply:
@nicholas-masel ok I flipped this one over.